test/rules/join_return_with_assignment.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N join_return_with_assignment`
+
+int bar() {
+  int a = 3;
+  --a; // LINT
+  return a;
+}
+
+int foo() {
+  int a = 3;
+  a++; // LINT
+  return a;
+}
+
+int doubleFoo() {
+  int a = 3;
+  a += 5;
+  a += 15; // OK
+  return a;
+}
```

Comment:
Do we want to produce a lint if there are multiple assignments but they are not next to each other, as in
```
int a = 3;
a += 5;
print('about to add 15');
a += 15;
return a;
```
This seems inconsistent. (Maybe I just don't understand why the number of assignments matters.)